editor/components/post-permalink/index.js

```diff
@@ -8,61 +8,93 @@ import { connect } from 'react-redux';
  */
 import { Component } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Dashicon, ClipboardButton, Button } from '@wordpress/components';
+import { Dashicon, Button, ClipboardButton, Tooltip } from '@wordpress/components';
 
 /**
  * Internal Dependencies
  */
 import './style.scss';
-import { isEditedPostNew, getEditedPostAttribute } from '../../store/selectors';
+import PostPermalinkEditor from './editor.js';
+import { isEditedPostNew, isPermalinkEditable, getEditedPostPreviewLink, getPermalink } from '../../store/selectors';
+import { getWPAdminURL } from '../../utils/url';
 
 class PostPermalink extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.state = {
-			showCopyConfirmation: false,
+			iconClass: '',
+			editingPermalink: false,
 		};
-		this.onCopy = this.onCopy.bind( this );
-		this.onFinishCopy = this.onFinishCopy.bind( this );
-	}
-
-	componentWillUnmount() {
-		clearTimeout( this.dismissCopyConfirmation );
 	}
 
-	onCopy() {
-		this.setState( {
-			showCopyConfirmation: true,
-		} );
-	}
-
-	onFinishCopy() {
-		this.setState( {
-			showCopyConfirmation: false,
-		} );
+	componentDidUpdate( prevProps, prevState ) {
+		// If we've just stopped editing the permalink, focus on the new permalink.
```

Comment:
I don't think we can do it this way. Because `permalinkButton` only exists when `editingPermalink` is `false`, we can't focus it until after the render that causes the `permalinkButton` to come back into existence. Trying to focus it at the same time as setting `editingPermalink` to `false` just gives an error.